pkg/operatorsource/configuring.go

```diff
@@ -58,7 +58,7 @@ type configuringReconciler struct {
 // If the corresponding CatalogSourceConfig object already exists
 // then no further action is taken.
 func (r *configuringReconciler) Reconcile(ctx context.Context, in *v1alpha1.OperatorSource) (out *v1alpha1.OperatorSource, nextPhase *v1alpha1.Phase, err error) {
-	if in.Status.CurrentPhase.Name != phase.Configuring {
+	if in.GetCurrentPhaseName() != phase.Configuring {
```

Comment:
You should introduce GetCurrentPhaseName() in this commit. So this commit will come before `[opsrc] Introduce a new phase: "Purging"`.